myst_nb/parser.py

```diff
@@ -24,15 +28,26 @@ def parse(self, inputstring, document):
         self.reporter = document.reporter
         self.config = self.default_config.copy()
         try:
-            source_path = document.settings._source # source file path
+            source_path = document.settings.env.doc2path(document.settings.env.docname)
             new_cfg = document.settings.env.config.myst_config
             self.config.update(new_cfg)
         except AttributeError:
             pass
 
         ntbk = nbf.reads(inputstring, nbf.NO_CONVERT)
-    
-        ntbk = cached_execution_and_merge(source_path, ntbk)
+
+        # Populate the outputs either with nbclient or a cache
+        path_cache = document.settings.env.config['jupyter_cache']
+        if path_cache is True:
+            path_cache = Path(document.settings.env.srcdir).joinpath('.jupyter_cache')
```

Comment:
Hey @choldgraf , does it make more sense to have the `.jupyter_cache` folder inside the source directory? My intuition was to have it at the same level as `build` and `source` folders, top level of the project folder. 

Or maybe, when I think more about it, inside the `build` directory as sphinx does for its doctree cache? Then `make clean` will also clear the cache.

Reply:
Hmmm, do we think that people will want "make clean" to clear the cache? If so, that could be a good idea to put it

Reply:
Maybe, but it also might be my familiarity bias speaking. Because we have the same thing in jupinx for clearing execution cache, following the sphinx philosophy of storing doctree cache there. 
Basically we have specific cleanses like `make clean-html` , `make clean-execute`.For a clean slate, we have `make clean`.

One other advantage I think can be distribution. The `build` folder as a whole can be distributed, which has outputs and cache both in it?